app/controllers/locations_controller.rb

```diff
@@ -43,8 +43,16 @@ def deactivate_confirm
   end
 
   def deactivate
-    if @location.location_deactivation_periods.create!(deactivation_date: params[:deactivation_date]) && reset_location_and_scheme_for_logs!
+    if @location.location_deactivation_periods.create!(deactivation_date: params[:deactivation_date])
+      logs = reset_location_and_scheme_for_logs!
```

Comment:
This technically removes `reset_location_...` from the conditional, but since it's using `update!` it already either returns true or throws, so behaviour should be identical.